src/main/java/com/github/dockerjava/jaxrs/TagImageCmdExec.java

```diff
@@ -18,16 +19,40 @@ public TagImageCmdExec(WebTarget baseResource) {
 
 	@Override
 	protected Void execute(TagImageCmd command) {
-		WebTarget webResource = getBaseResource().path("/images/" + command.getImageId() + "/tag")
-                .queryParam("repo", command.getRepository())
-                .queryParam("tag", command.getTag())
+		WebTarget webResource = getBaseResource().path("/images/" + command.getOriginalImageName() + "/tag")
+                .queryParam("repo", getImageNameWithoutTag(command.getNewImageName()))
+                .queryParam("tag", command.getNewImageName().getTag())
                 .queryParam("force", command.hasForceEnabled() ? "1" : "0");
 
 		LOGGER.trace("POST: {}", webResource);
 		webResource.request().post(null).close();
 		return null;
 	}
-	
+
+	/**
+	 * Returns a String with all parts except Tag
+	 * @param imageName
+	 * @return
+	 */
+	private String getImageNameWithoutTag(DockerImageName imageName)
```

Comment:
In this case I'm unsure of the Remote API, it seems to want the new name split into repo (all but the tag) and the tag.